### <a name="systemthreadingtaskstask-no-longer-throw-objectdisposedexception-after-object-is-disposed"></a><span data-ttu-id="4c468-101">System.Threading.Tasks.Task は、オブジェクトが破棄された後に不要になった ObjectDisposedException をスローします。</span><span class="sxs-lookup"><span data-stu-id="4c468-101">System.Threading.Tasks.Task no longer throw ObjectDisposedException after object is disposed</span></span>

|   |   |
|---|---|
|<span data-ttu-id="4c468-102">説明</span><span class="sxs-lookup"><span data-stu-id="4c468-102">Details</span></span>|<span data-ttu-id="4c468-103">除く<xref:System.Threading.Tasks.Task.System%23IAsyncResult%23AsyncWaitHandle>、<xref:System.Threading.Tasks.Task?displayProperty=name>メソッドをスローしなく、<xref:System.ObjectDisposedException?displayProperty=name>例外オブジェクトが破棄された後にします。この変更は、キャッシュされたタスクの使用をサポートします。</span><span class="sxs-lookup"><span data-stu-id="4c468-103">Except for <xref:System.Threading.Tasks.Task.System%23IAsyncResult%23AsyncWaitHandle>, <xref:System.Threading.Tasks.Task?displayProperty=name> methods no longer throw an <xref:System.ObjectDisposedException?displayProperty=name> exception after the object is disposed.This change supports the use of cached tasks.</span></span> <span data-ttu-id="4c468-104">たとえば、メソッドで新しいタスクを割り当てる代わりに、既に完了した操作を表す、キャッシュされたタスクを返すことができます。</span><span class="sxs-lookup"><span data-stu-id="4c468-104">For example, a method can return a cached task to represent an already completed operation instead of allocating a new task.</span></span> <span data-ttu-id="4c468-105">これは、タスクの任意のコンシューマーによって破棄されてしまうと、使用不可能になってしまう以前の .NET Framework のバージョンでは不可能でした。</span><span class="sxs-lookup"><span data-stu-id="4c468-105">This was impossible in previous .NET Framework versions, because any consumer of the task could dispose of it, which rendered it unusable.</span></span>|
|<span data-ttu-id="4c468-106">提案される解決策</span><span class="sxs-lookup"><span data-stu-id="4c468-106">Suggestion</span></span>|<span data-ttu-id="4c468-107">タスク メソッドをスローしない可能性がありますに注意してください<xref:System.ObjectDisposedException?displayProperty=name>場合、オブジェクトが破棄されるときにします。</span><span class="sxs-lookup"><span data-stu-id="4c468-107">Be aware that Task methods may no longer throw <xref:System.ObjectDisposedException?displayProperty=name> in cases when the object is disposed.</span></span> <span data-ttu-id="4c468-108">タスクの状態を明示的にチェックを更新する必要がある場合は、アプリによってされたタスクが破棄されたことを確認するには、この例外を使用して<xref:System.Threading.Tasks.Task.Status>です。</span><span class="sxs-lookup"><span data-stu-id="4c468-108">If an app was depending on this exception to know that a task was disposed, it should be updated to explicitly check the task's status using <xref:System.Threading.Tasks.Task.Status>.</span></span>|
|<span data-ttu-id="4c468-109">スコープ</span><span class="sxs-lookup"><span data-stu-id="4c468-109">Scope</span></span>|<span data-ttu-id="4c468-110">マイナー</span><span class="sxs-lookup"><span data-stu-id="4c468-110">Minor</span></span>|
|<span data-ttu-id="4c468-111">Version</span><span class="sxs-lookup"><span data-stu-id="4c468-111">Version</span></span>|<span data-ttu-id="4c468-112">4.5</span><span class="sxs-lookup"><span data-stu-id="4c468-112">4.5</span></span>|
|<span data-ttu-id="4c468-113">型</span><span class="sxs-lookup"><span data-stu-id="4c468-113">Type</span></span>|<span data-ttu-id="4c468-114">ランタイム</span><span class="sxs-lookup"><span data-stu-id="4c468-114">Runtime</span></span>|

